raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -360,26 +359,30 @@ private void maybeBecomeLeader(CandidateState state) throws IOException {
         }
     }
 
-    private void onBecomeCandidate(CandidateState state) throws IOException {
+    private void onBecomeCandidate(CandidateState state, int retries) throws IOException {
         maybeBecomeLeader(state);
         resetConnections();
 
         kafkaRaftMetrics.updateElectionStartMs(time.milliseconds());
-        timer.reset(electionTimeoutMs + randomElectionJitterMs());
+        timer.reset(randomElectionTimeoutMs());
     }
 
-    private void becomeCandidate() throws IOException {
+    private void becomeCandidate(int retries) throws IOException {
```

Comment:
It's a little awkward to pass around `retries` like this. Not sure I follow why it's necessary since we keep track of retries in `CandidateState`.

Reply:
Ok, I think I understand the issue. Basically the `CandidateState` is constructed afresh on every new election attempt. So we don't have a natural way to carry `retries` over. But maybe we can do a little better and push this into `QuorumState.becomeCandidate`. If we detect that we are already a candidate, then we can pass through `retries`. At least that lets us keep the state out of `KafkaRaftClient`.

Reply:
Good idea. Will try it out.